src/libstd/sys/wasm/mutex_atomics.rs

```diff
@@ -26,26 +26,30 @@ impl Mutex {
 
     pub unsafe fn lock(&self) {
         while !self.try_lock() {
-            let val = wasm32::i32_atomic_wait(
-                self.ptr(),
-                1,  // we expect our mutex is locked
-                -1, // wait infinitely
-            );
+            // SAFETY: the caller must uphold the safety contract for `i32_atomic_wait`.
```

Comment:
This seems... odd. @alexcrichton, can you recall why lock() here is unsafe? It seems like it could be safe.

Reply:
The intrinsic has a `*mut i32` in the signature which it modifies, so the intrinsic is unsafe.